src/wayland/buffer.rs

```diff
@@ -15,6 +14,49 @@ use wayland_client::{
 
 use super::State;
 
+#[cfg(any(target_os = "linux", target_os = "freebsd"))]
+fn create_memfile() -> File {
+    use nix::sys::memfd::{memfd_create, MemFdCreateFlag};
```

Comment:
I generally prefer `rustix` to `nix` when available, but given that most of `softbuffer`'s downstream dependencies use `nix` anyways, I'm fine with this for now.

Reply:
Yeah, I found out about `rustix` recently and it could be a better choice. Though nix is indeed likely to be a dependency of other things. (Though this only matters if they depend on the same version of nix, anyway.)

Reply:
At the moment, most of the Wayland crates depend on `nix`, as well as `x11rb`. For the time being we probably should just stick with `nix`.

Reply:
Rustix is also less complete. It looks like it doesn't currently have wrappers around `shm_open` and `shm_unlink`.

Of course the minimal solution is just to use the libc crate.